OWML.ModHelper.Interaction/InterfaceProxyFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text;
+
+namespace OWML.ModHelper.Interaction
+{
+    public class InterfaceProxyFactory
```

Comment:
Same as above, I'd like it to be 100% clear this is based on SMAPI code :)

Reply:
done!